modules/lightning_telemetry/src/Telemetry.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Drupal\lightning_telemetry;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleExtensionList;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Telemetry service.
+ */
+class Telemetry {
+
+  /**
+   * Amplitude API URL.
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  const AMPLITUDE_API_URL = 'https://api.amplitude.com/httpapi';
+
+  /**
+   * Amplitude API key.
+   *
+   * This is not intended to be private. It is typically included in client
+   * side code. Fetching data requires an additional API secret.
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  const AMPLITUDE_API_KEY = 'f32aacddde42ad34f5a3078a621f37a9';
+
+  /**
+   * The extension.list.module service.
+   *
+   * @var \Drupal\Core\Extension\ModuleExtensionList
+   */
+  protected $extensionListModule;
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * The config.factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $config_factory;
+
+  /**
+   * Constructs a telemetry object.
+   *
+   * @param \Drupal\Core\Extension\ModuleExtensionList $extension_list_module
+   *   The extension.list.module service.
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config.factory service.
+   */
+  public function __construct(ModuleExtensionList $extension_list_module, ClientInterface $http_client, ConfigFactoryInterface $config_factory) {
+    $this->extensionListModule = $extension_list_module;
+    $this->httpClient = $http_client;
+    $this->config_factory = $config_factory;
+  }
+
+  /**
+   * Sends an event to Amplitude.
+   *
+   * @param \Drupal\lightning_telemetry\Event $event
+   *   The Amplitude event.
+   *
+   * @return bool
+   *   TRUE if the request to Amplitude was successful, FALSE otherwise.
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  protected function sendEvent(Event $event) {
+    $response = $this->httpClient->request('POST', self::AMPLITUDE_API_URL, [
+      'form_params' => [
+        'api_key' => self::AMPLITUDE_API_KEY,
+        'event' => Json::encode($event),
+      ]
+    ]);
+
+    return $response->getStatusCode() == 200;
+  }
+
+  /**
+   * Creates and sends a cron event to Amplitude.
+   *
+   * @param string $event_type
+   *   The event type.
+   *
+   * @param array $event_properties
+   *   Event properties.
+   *
+   * @throws \Exception
+   */
+  public function sendTelemetry($event_type, array $event_properties = []) {
```

Comment:
If we allow arbitrary properties to be sent, anybody could use this method to populate our metrics with garbage. We probably need to consider a properties whitelist approach.

Reply:
...garbage _values_ to whitelisted properties could be a problem, too. Depending on our level of concern, we might need to be even more aggressive, such as type testing values or somehow preventing non-Acquia code from sending metrics at all.

Reply:
I'm not very concerned with receiving garbage values. Amplitude provides a way to ignore these when/if it happens. I'm not sure this warrants the effort.